src/zarr/core/metadata/v2.py

```diff
@@ -332,6 +367,18 @@ def parse_fill_value(fill_value: object, dtype: np.dtype[Any]) -> Any:
             raise ValueError(
                 f"fill_value {fill_value!r} is not valid for dtype {dtype}; must be a unicode string"
             )
+    elif dtype.kind in "SV" and isinstance(fill_value, str):
+        fill_value = base64.standard_b64decode(fill_value)
```

Comment:
out of scope but does anyone know why we base64 encode scalars with dtype `S` ? `S` is ascii-encoded strings, which shouldn't need any encoding / decoding.